packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -89,12 +89,24 @@ export function navigate<RouteName extends keyof StackParamList>(
     .then(() => {
       Logger.debug(`${TAG}@navigate`, `Dispatch ${routeName}`)
 
-      navigationRef.current?.dispatch(
-        NavigationActions.navigate({
-          routeName,
-          params,
-        })
-      )
+      // Remove the WC loading screen from the stack as we never want to land
+      // on this screen when navigating back. Ensures that the dapp webview WC
+      // navigation works correctly.
+      if (navigationRef.current?.getCurrentRoute()?.name === Screens.WalletConnectLoading) {
+        navigationRef.current?.dispatch(
+          StackActions.replace({
+            routeName,
+            params,
+          })
+        )
```

Comment:
Wondering, how does it look visually when it's replaced?
Is it a hard replace, without any animation?

Reply:
really good question...i've been playing around with it all afternoon and not sure what to do with it (couldn't find the right combination of props to make it perfect). it seems that the replace has some animation but it's not quite right and the screen is removed before the next screen fully replaces it. on most dapps this is fine, but for the dark/colorful ones you can see a flash:

https://user-images.githubusercontent.com/20150449/159040200-817961f5-2c98-490f-9cec-96aad7822e66.mov

i'll keep looking into it but also wondered how much on a scale of 1-10 this triggers your OCD? 😅

Reply:
That's fine lol. The fullscreen signing stuff is only temporary 😄 
(well everything temporary stays for a while in reality).